.gitlab-ci.yml

```diff
@@ -15,6 +15,8 @@ include:
 
 stages:
   - build
+  - publish_python_image
+  - publish_job_base_image
   - publish_artifacts
```

Comment:
Is there reason for adding new stags. Why not use either publish_artifacts stage? 

If there are some specific dependencies we can use https://docs.gitlab.com/ee/ci/yaml/#needs

Since stages are global for all projects CI I would like us to have good justification for adding or changing them.
Since more stages make CICD harder to understand for people. 

 Also the name of the stages are very specific for the Control Service. I think needs is the better solution here if dependencies are required.

Reply:
Agreed, I will switch to the `needs`, thanks.

Reply:
I was going to suggest this initially, but seeing the amount of jobs that would have to be chained with `needs` seemed like it would make the configuration file too verbose. 😕 I'm ok with whatever you decide @mivanov1988 , just think about this trade-off as well.